crates/sdk/src/env.rs

```diff
@@ -181,3 +184,47 @@ pub fn state_write<T: AppState>(state: &T) {
     };
     let _ = storage_write(STATE_KEY, &data);
 }
+
+/// Generate a new random UUID v4.
+#[inline]
+#[must_use]
+pub fn generate_uuid() -> Uuid {
+    #[cfg(target_arch = "wasm32")]
+    {
+        unsafe { sys::generate_uuid(DATA_REGISTER) };
+        Uuid::from_slice(&read_register_sized::<16>(DATA_REGISTER).expect("Must have UUID"))
+            .expect("UUID must be valid")
+    }
+    #[cfg(not(target_arch = "wasm32"))]
+    {
+        Uuid::new_v4()
+    }
```

Comment:
these are not necessary, anything that would use `generate_uuid` would've already called (at least) `state_read -> storage_read`, which would according to https://github.com/calimero-network/core/blob/54336ece92deffa11447c54e1f420fb959f7d727/crates/sdk/src/sys.rs#L48-L50 simply say "not supported"

the same would be maintained of `sys::generate_uuid`, so the mocks are not needed

Reply:
See below example with `time_now()`, as that's unaffected by the wider question around UUID type. For this specific context, the same applies - i.e. cannot remove this.

Reply:
Replied below